controllers:
  chassis_controller:
    type: rm_chassis_controllers/BalanceController
    # ChassisBase
    publish_rate: 100
    enable_odom_tf: true
    power:
      coeff: 0.535
      min_vel: 4.4
    timeout: 0.1
    pid_follow: { p: 5, i: 0, d: 0.05, i_max: 0.0, i_min: 0.0, antiwindup: true, publish_state: true }
    twist_covariance_diagonal: [ 0.001, 0.001, 0.001, 0.001, 0.001, 0.001 ]
    # BalanceController
    joint_left_name: "left_wheel_joint"
    joint_right_name: "right_wheel_joint"
    wheel_radius: 0.06
    wheel_base: 0.405
    com_pitch_offset: 0.05
    a: [ [ 0, 1, 0, 0 ],
         [ 21.7204, 0, 0, 0 ],
         [ 12.4948, 0, 0, 0 ],
         [ 0, 0, 0, 0 ] ]
    b: [ [ 0, 0 ],
         [ -64.4355, -64.4355 ],
         [ 7.7211, 7.7211 ],
         [ -7.1659, 7.1659 ] ]
    q: [ [ 0.01, 0, 0, 0 ],
         [ 0, 0.01, 0, 0 ],
         [ 0, 0, 0.1, 0 ],
         [ 0, 0, 0, 0.01 ] ]
    r: [ [ 1, 0 ],
         [ 0, 1 ] ]
